### PR TITLE
Creates new AWS Ruby SDK api commands and tests

### PR DESCRIPTION
**Pull request by ebozduman**
Since Minio Ruby SDK for Amazon S3 is not available yet, I wrote the basic functionality as part of the Mint testing effort. Then created test cases which basically call these commands and check if the expected values are returned.
I took Minio Java SDK for Amazon S3 list of command as a base, and created Ruby SDK commands using the same names.
Here is the list of commands, for which tests created, in this pull request.
Bucket Operations:
- makeBucket
- listBuckets
- bucketExists?
- removeBucket
- listObjects

Object Operations:
- getObject
- putObject
- copyObject
- statObject
- removeObjects
- removeObject

### CODE REVIEW

**Comment by NitishT** [comment]
Tabbing is off @ebozduman . Need to probably run the lint tool

**Comment by NitishT** [comment]
any reviews here @harshavardhana

File: run/core/aws-sdk-ruby/aws-stub-tests.rb [hunk 1]
```diff
@@ -62,182 +61,459 @@ def print_status(result, e='')
...
 data_dir = ENV['MINT_DATA_DIR'] ||= 'MINT_DATA_DIR is not set'
-
 # "1/0" value to decide if "HTTPS"
 # needs to be used on or not.
 enable_https = ENV['ENABLE_HTTPS']
-
 # Add "https://" to "endpoint" if environment
-# variable "ENABLE_HTTPS" is turned on
+# variable "ENABLE_HTTPS" is set to 1
 endpoint = enable_https == "1" ? 'https://' + endpoint : 'http://' + endpoint
```
Comment (harshavardhana) [inline run/core/aws-sdk-ruby/aws-stub-tests.rb]: isn't it possible to do 

```
endpoint = ENV['ENABLE_HTTPS'] == "1" ? 'https://' + endpoint : 'http://' + endpoint
``` 

Is `enable_https` used anywhere?

Comment (ebozduman) [inline run/core/aws-sdk-ruby/aws-stub-tests.rb]: Harsha,
It is already that way in the latest code. Maybe you  see the older version of the file.
Could you please recheck it?

**Pull request merged**